### PR TITLE
Update changelog for 1.4.1

### PR DESCRIPTION
**Pull request by deepakkinni**
**What this PR does / why we need it**:
Changelog update for 1.4.1

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note

```
Signed-off-by: Deepak Kinni <dkinni@vmware.com>

### CODE REVIEW

File: changelogs/CHANGELOG-1.4.md [hunk 1]
```diff
@@ -1,5 +1,307 @@
...
 
+## v1.4.1
+
+Date: 2022-11-8
+
+### Changes
+
+- Add 1.4.0 to readme compatibility matrix. (#480, @xinyanw409)
+- User photon image and update dependencies (#491, @xing-yang)
+-
```
Comment (xing-yang) [inline changelogs/CHANGELOG-1.4.md:11]: Remove "-"

File: changelogs/CHANGELOG-1.4.md [hunk 2]
```diff
@@ -1,5 +1,307 @@
 # Changelog since v1.3.0
 
+## v1.4.1
+
+Date: 2022-11-8
+
+### Changes
+
+- Add 1.4.0 to readme compatibility matrix. (#480, @xinyanw409)
```
Comment (xing-yang) [inline changelogs/CHANGELOG-1.4.md:9]: Can you make PR and gitid hyperlink like in this one? https://github.com/kubernetes-csi/external-snapshotter/blob/master/CHANGELOG/CHANGELOG-6.1.md#api-change

File: changelogs/CHANGELOG-1.4.md [hunk 3]
```diff
@@ -1,5 +1,307 @@
 # Changelog since v1.3.0
```
Comment (xing-yang) [inline changelogs/CHANGELOG-1.4.md:1]: Shouldn't this be Changelog since 1.4.0?

**Comment by svcbot-qecnsdp** [comment]
Sorry, some of the pre-check tests are failing for this PR ! 

Please check the Jenkins job for more details
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/758/

File: changelogs/CHANGELOG-1.4.md (hunk 3 above)
Comment (deepakkinni) [reply changelogs/CHANGELOG-1.4.md:1]: 1.4.0 changelog describes all changes 1.3.0, it's how CHANGELOG-1.1 and CHANGELOG-1.2 were written.

**Pull request merged**